src/osd/OSD.cc

```diff
@@ -8656,7 +8656,7 @@ void OSD::handle_pg_query(OpRequestRef op)
       peering_wait_for_split[pgid].push_back(
 	PG::CephPeeringEvtRef(
 	  new PG::CephPeeringEvt(
-	    it->second.epoch_sent, it->second.epoch_sent,
+	    m->get_epoch(), it->second.epoch_sent,
```

Comment:
i don't think we should use the osdmap's epoch of the *OSD* when the `MOSDPGQuery` is sent as the pg's `sent_epoch`. because this is all about PG. even if `m->get_epoch()` is newer than the one possessed by the recipient PG , the recipient should still be able to handle it as long as its osdmap's epoch is greater or equal to `it->second.epoch_sent`.

Reply:
Maybe you are right, but the declaration of PG::queue_query() really puzzles me.

Reply:
well, IMHO, `msg_epoch` is different from `current_map_epoch`. if you have a better idea. please feel free to send a PR.